File: benchmarks/executable/process_arguments_count.c
// C23 reference for the public process-arguments-count executable workload.

#include <stdio.h>

#ifdef _WIN32
#include <fcntl.h>
#include <io.h>
#endif

int main(int argc, char **argv) {
    (void)argv;

#ifdef _WIN32
    if (_setmode(_fileno(stdout), _O_BINARY) == -1) return 1;
#endif

    const int count = argc - 1;
    if (count == 2) {
        if (fputs("Exactly two arguments\n", stdout) == EOF) return 1;
    } else if (printf("Argument count %d\n", count) < 0) {
        return 1;
    }
    return 0;
}

File: benchmarks/executable/process_arguments_count.rs
// Rust 2024 reference for the public process-arguments-count executable workload.

use std::io::{self, Write};

fn main() {
    let count = std::env::args_os().count().saturating_sub(1);
    let mut stdout = io::stdout().lock();
    let written = if count == 2 {
        writeln!(stdout, "Exactly two arguments")
    } else {
        writeln!(stdout, "Argument count {count}")
    };
    if written.is_err() {
        std::process::exit(1);
    }
}
